src/main/java/org/broadinstitute/hellbender/engine/TwoPassVariantWalker.java

```diff
@@ -0,0 +1,70 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import org.broadinstitute.hellbender.engine.filters.CountingReadFilter;
+import org.broadinstitute.hellbender.engine.filters.VariantFilter;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+
+import java.util.stream.StreamSupport;
+
+public abstract class TwoPassVariantWalker extends VariantWalker {
```

Comment:
For tests, I think we need 3 things here:

* A `TwoPassVariantWalkerUnitTest` test class, modeled after `TwoPassReadWalkerUnitTest`
* An `ExampleTwoPassVariantWalker` tool in `tools.examples`, modeled after the existing `ExampleVariantWalker` tool
* A simple `ExampleTwoPassVariantWalkerIntegrationTest`, modeled after `ExampleVariantWalkerIntegrationTest`

Reply:
The unit test for `TwoPassVariantWalker` writes a private example walker class and runs it, which seems to be the same as the integration test. I think doing either is sufficient - what do you think?

Reply:
It's good to have both the unit test and the integration test. The unit test uses a toy implementation to count calls to first/second pass apply, which is a good correctness test for the traversal. However, we also try to include a runnable example implementation that actually produces meaningful output for every walker type as a template for tool developers, and that's where `ExampleTwoPassVariantWalker` comes in.